src/components/ProjectPage/ProjectPage.js

```diff
@@ -58,7 +62,11 @@ class ProjectPage extends PureComponent<Props> {
   }
 
   componentWillReceiveProps(nextProps: Props) {
-    if (this.props.project.id !== nextProps.project.id) {
+    const { project } = this.props;
+    if (
+      project.id !== nextProps.project.id ||
+      project.dependencies.length === 0
```

Comment:
Not exactly sure if this caused the looping. But I'll try if it can be removed. I think a update is required if there are now dependencies in state so reload from disk should be triggered. So I think it can't be removed but maybe we can only reload if `loadedStatus` in `dependencies.reducer` is `loading`. Maybe we have to add a state like `in progress` so this can be prevented from retriggering loading.

For me the looping stopped by removing the swallowing of the EONET error. But if you're still having it I'll check that line.

Reply:
OK, I've removed the `project.dependencies.length` check and couldn't detect a difference. Or am I missing a case? 
Please check if the issue is fixed with this.

Reply:
Works great now :-) Yay! 

RE: updating if there are new dependencies in state. From how I understand the new code, triggering a reinstall via clicking the button should trigger the reload from disk/UI update from this code in dependency.saga.js:

```js
// load dependencies to refresh state
yield put(loadDependencyInfoFromDiskStart(projectId, projectPath));
```

But @joshwcomeau reports things still not working for him properly so maybe I'm wrong.